### DISCUSSION TITLE
Count the number of relationships

### DISCUSSION THREAD

**Discussion by alexandrepletty** [directus/directus#14047] [Archive]
Hello, is it possible to count the number of objects in a relationship according to its status?

Example, an article has two categories and one of them has published status and the second has draft status, how can I do when I do count(categories) it counts one instead of two.

thank you in advance :)

**Comment by br41nslug** [comment]
You'll want to add a `deep` filter to filter out the draft statuses like `?deep={"categories":{"_filter":{"status":{"_eq":"published"}}}}}` https://docs.directus.io/reference/query/#examples-7

**Reply by alexandrepletty** [reply]
Yes but when I put in fields=count(categories) and I filter it with deep, it always returns me 2 instead of 1

**Reply by br41nslug** [reply]
Strange if you use `fields=*.*` instead does it filter the categories like expected?

**Reply by alexandrepletty** [reply]
categories_count always remains at 2 instead of 1 🙄

**Reply by SeanDylanGoff** [reply]
I think the problem is you're counting categories not articles. the endpoint has to be /items/articles and the count could be * or id or whatever.

**Reply by alexandrepletty** [reply]
I select the articles with their categories and I would like to display only the categories which have the status published, but I use count in the field to be able to select the number of categories but it does not make the difference between the published and the drafts
Endpoint : `/items/article`
field parameter : `count(categories),categories.categories_id.*`
Deep Parameter : `{"categories": {"_filter":{"categories": {"status": { "_eq": "published"}}}}}`
return :
`
{
	"data": [
		{
			"categories_count": "0",
			"categories": []
		},
		{
			"categories_count": "2",
			"categories": [
				{
					"categories_id": {
						"id": "fb525ad3-ab0e-41e8-a8de-e4cac457de18",
						"status": "published",
						"user_created": "594a97f4-b2ac-4b86-abfc-97eec693dedc",
						"date_created": "2022-06-19T17:09:39.053Z",
						"content": "nice",
						"article": [
							2
						]
					}
				}
			]
		},
		{
			"categories_count": "0",
			"categories": []
		}
	]
}
`
The filter select 1 published item but the count give me all and I would like to select only the number that is filtered 😄

**Reply by SeanDylanGoff** [reply]
Could you/did you try `count(categories.categories_id.id),categories.categories_id.*`, `count(categories.categories_id.status),categories.categories_id.*` and `count(categories.categories_id),categories.categories_id.*`?

**Reply by alexandrepletty** [reply]
it removes "comments_count"
